File: SDKs/java/src/main/java/com/intelliversex/sdk/ai/IVXAIVoiceServices.java
// Copyright (c) 2026 Intelli-verse-X
// MIT License — see LICENSE in the project root.

package com.intelliversex.sdk.ai;

import java.util.List;
import java.util.concurrent.CompletableFuture;

/**
 * Speech-to-text, text-to-speech, voices, streaming STT (Unity {@code IVXAIVoiceServices}).
 */
public final class IVXAIVoiceServices {

    private static final IVXAIVoiceServices INSTANCE = new IVXAIVoiceServices();

    private IVXAIVoiceServices() {}

    public static IVXAIVoiceServices getInstance() {
        return INSTANCE;
    }

    public boolean isTranscribing() {
        return false;
    }

    public List<IVXAIVoice> getAvailableVoices() {
        throw new UnsupportedOperationException("Not yet implemented — stub only");
    }

    public void initialize(Object config) {
        throw new UnsupportedOperationException("Not yet implemented — stub only");
    }

    public CompletableFuture<IVXTranscriptionResult> transcribeAudio(byte[] pcmData, int sampleRate) {
        return failed();
    }

    public CompletableFuture<byte[]> synthesizeSpeech(String text, String voiceId) {
        return failed();
    }

    public CompletableFuture<List<IVXAIVoice>> listVoices() {
        return failed();
    }

    public CompletableFuture<IVXLanguageDetectResult> detectLanguage(byte[] pcmData, int sampleRate) {
        return failed();
    }

    public void startStreamingTranscription(int sampleRate) {
        throw new UnsupportedOperationException("Not yet implemented — stub only");
    }

    public void stopStreamingTranscription() {
        throw new UnsupportedOperationException("Not yet implemented — stub only");
    }

    public void feedAudioChunk(byte[] pcmChunk) {
        throw new UnsupportedOperationException("Not yet implemented — stub only");
    }

    private static <T> CompletableFuture<T> failed() {
        CompletableFuture<T> f = new CompletableFuture<>();
        f.completeExceptionally(new UnsupportedOperationException("Not yet implemented — stub only"));
        return f;
    }

    public static final class IVXAIVoice {
        public String voiceId;
        public String displayName;
        public String language;
        public String gender;
        public String previewUrl;
        public String[] tags;
    }

    public static final class IVXTranscriptionResult {
        public String text;
        public String language;
        public float confidence;
        public boolean isFinal;
    }

    public static final class IVXLanguageDetectResult {
        public String language;
        public float confidence;
    }
}

File: SDKs/cocos2dx/Classes/IntelliVerseX/IVXAIVoiceServices.cpp
// Copyright (c) 2026 Intelli-verse-X
// MIT License — see LICENSE in the project root.

#include "IntelliVerseX/IVXAIVoiceServices.h"
#include "cocos2d.h"
#include <stdexcept>

namespace IntelliVerseX {

IVXAIVoiceServices& IVXAIVoiceServices::getInstance() {
    static IVXAIVoiceServices instance;
    return instance;
}

bool IVXAIVoiceServices::isTranscribing() const {
    return false;
}

void IVXAIVoiceServices::initialize(void*) {
    cocos2d::log("[IVX-Cocos] IVXAIVoiceServices::initialize: stub — not yet implemented. AI features will return empty results.");
}

void IVXAIVoiceServices::transcribeAudio(const std::uint8_t*, std::size_t, int,
                                          std::function<void(const IVXTranscriptionResult&)>, ErrorCallback onError) {
    cocos2d::log("%s", "[IVX-Cocos] AI stub: not yet implemented — stub only");
    if (onError) onError({-1, "Not yet implemented — stub only"});
}

void IVXAIVoiceServices::synthesizeSpeech(const std::string&, const std::string&,
                                          std::function<void(const std::vector<std::uint8_t>&)>, ErrorCallback onError) {
    cocos2d::log("%s", "[IVX-Cocos] AI stub: not yet implemented — stub only");
    if (onError) onError({-1, "Not yet implemented — stub only"});
}

void IVXAIVoiceServices::listVoices(std::function<void(const std::vector<IVXAIVoice>&)>, ErrorCallback onError) {
    cocos2d::log("%s", "[IVX-Cocos] AI stub: not yet implemented — stub only");
    if (onError) onError({-1, "Not yet implemented — stub only"});
}

void IVXAIVoiceServices::detectLanguage(const std::uint8_t*, std::size_t, int,
                                        std::function<void(const std::string&, float)>, ErrorCallback onError) {
    cocos2d::log("%s", "[IVX-Cocos] AI stub: not yet implemented — stub only");
    if (onError) onError({-1, "Not yet implemented — stub only"});
}

void IVXAIVoiceServices::startStreamingTranscription(int) {
    cocos2d::log("%s", "[IVX-Cocos] AI stub: not yet implemented — stub only");
    throw std::runtime_error("Not yet implemented — stub only");
}

void IVXAIVoiceServices::stopStreamingTranscription() {
    cocos2d::log("%s", "[IVX-Cocos] AI stub: not yet implemented — stub only");
    throw std::runtime_error("Not yet implemented — stub only");
}

void IVXAIVoiceServices::feedAudioChunk(const std::uint8_t*, std::size_t) {
    cocos2d::log("%s", "[IVX-Cocos] AI stub: not yet implemented — stub only");
    throw std::runtime_error("Not yet implemented — stub only");
}

} // namespace IntelliVerseX
